### PR TITLE
feat: implement create checkout hook for multi ship

### PR DESCRIPTION
**Pull request by kibo-sahilarora**

### CODE REVIEW

File: __mocks__/stories/multiShipCheckoutMock.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import type { Checkout } from '@/lib/gql/types'
+
+const multiShipCheckoutResponse: Checkout = {
```
Comment (kibo-sushant) [inline __mocks__/stories/multiShipCheckoutMock.ts:3]: rename the file name to checkoutMock (We have used similar conventions for other file names)

**Review by TriptiVermaXP — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        deliveryInstructions
+      }
+      availableActions
+      data
+      taxData
+      channelCode
+      locationCode
+      ipAddress
+      sourceDevice
+      visitId
```
Comment (kibo-kevinwatts) [inline lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts:98]: remove 
```      ipAddress
      sourceDevice
      visitId
      webSessionId
      customerInteractionType```

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      customerTaxId
+      isTaxExempt
+      currencyCode
+      priceListCode
+      attributes {
+        fullyQualifiedName
+        attributeDefinitionId
+        values
+      }
+      shopperNotes {
```
Comment (kibo-kevinwatts) [inline lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts:86]: remove shopperNotes

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      }
+      payments {
+        ...checkoutPaymentFragment
+      }
+      amountRemainingForPayment
+      acceptsMarketing
+      customerAccountId
+      email
+      customerTaxId
+      isTaxExempt
```
Comment (kibo-kevinwatts) [inline lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts:78]: remove ```   customerTaxId
      isTaxExempt```

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    createCheckout(cartId: $cartId) {
+      id
+      originalCartId
+      submittedDate
+      items {
+        destinationId
+        ...checkoutLineItemFragment
+      }
+      groupings {
+        id
```
Comment (kibo-kevinwatts) [inline lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts:18]: Grouping properties should be a fragment

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
+import {
```
Comment (kibo-kevinwatts) [inline lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts:1]: we should move this to a different folder, "multiship-checkout" or something to keep them separate from existing checkout mutations

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts (hunk 2 above)
Comment (kibo-sahilarora) [reply lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts]: Done

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts (hunk 3 above)
Comment (kibo-sahilarora) [reply lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts]: Done

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts (hunk 4 above)
Comment (kibo-sahilarora) [reply lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts]: Done

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts (hunk 5 above)
Comment (kibo-sahilarora) [reply lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts]: Done

File: lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts (hunk 6 above)
Comment (kibo-sahilarora) [reply lib/gql/mutations/checkout/create-multi-ship-checkout-mutation.ts]: Done

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**